.github/workflows/client.yml

```
@@ -0,0 +1,81 @@
+name: Go
+
+#TODO: extend the conditions once workflow gets tested together with other workflows
+on:
+  push:
+    branches:
+      # TODO: Run on master after we're fully migrated from Circle CI
+      # - master
+      - "rfc-18/**"
+  pull_request:
+      branches:
+      # TODO: Run on all branches after we're fully migrated from Circle CI
+      - "rfc-18/**"
+  workflow_dispatch:
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+      # TODO: This step was left here intentionally so we can track disk space
+      # usage for a while. We were trying to fight problems with out of disk space
+      # that happened due to the size of data restored from cache. The cache size
+      # was growing linearly with subsequent workflow runs. We want to observe
+      # available disk space for `/`. Fresh execution starts with 21 GB, we expect
+      # to have no less than 15 GB after the cache is restored.
+      - run: sudo df -h
+
+      - name: Build Docker Build Image
+        uses: docker/build-push-action@v2
+        with:
+          target: gobuild
+          tags: go-build-env
+          build-args: |
+            REVISION=${{ github.sha }}
```

I know I personally added it in keep-core, but can we comment it out. After consideration I think we shouldn't pass this argument during work on feature branches but only after the merge to master when we want to publish an image. Updating revision on each commit on feature branch causes longer docker image builds as it won't restore go client layer from cache but will rebuild it every time.
```suggestion
            # REVISION=${{ github.sha }} TODO: set revision only when we want to publish the image
```